code-challenges.test.js

```diff
@@ -14,33 +14,137 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const people = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+describe("nameJob", () => {
+	it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", () => {
+		const people = [
+			{ name: "ford prefect", occupation: "a hitchhiker" },
+			{ name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+			{ name: "arthur dent", occupation: "a radio employee" },
+		]
+		// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+		expect(nameJob(people)).toEqual([
+			"Ford Prefect is a hitchhiker.",
+			"Zaphod Beeblebrox is president of the galaxy.",
+			"Arthur Dent is a radio employee.",
+		])
+	})
+})
+
+// FAILED:  ReferenceError: nameJob is not defined
 
 // b) Create the function that makes the test pass.
 
+// Pseudo
+
+// Input: array of objects
+// Output: array with a sentence about each person with their name capitalized.
+// Expected: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+
+// Process
+
+// Declare a function called nameJob with the paramater array
+// My first thought is to use .map() to iterate through the object and return only the values of each object
+// Use the keys { name, occupation } to return the values
+// For name we need to split(" ") into two seperate words ex.) "ford", "perfect"
+// Then map again to go through the name "strings" and select character at the 0 index so we can use toUpperCase
+// Once the name strings are capitalized we can remove the original lowercase character with slice at the 1st position and it will be replaced by the capitalized value followed by the rest of the word.
+// Then we join them back in to one string.
+// I can use the + sign to include the string " is " the occupation value of the object, and finally the period "."
+// If I return the entire chain I should see the desired outcome
```

Comment:
- Nice work making sure your pseudo code covered every step of your logic!
- Good use of destructuring the object
- It would probably be easier to read and more efficient overall to use string interpolation rather than chaining a bunch of concatenations

Reply:
Thanks Chris, I was trying to use string interpolation in the beginning, but I gave up because it kept returning "undefined is a undefined". I will have to go back and see if I can refactor the string interpolation in.